autogen/agentchat/contrib/graph_rag/graph_rag_agent.py

```diff
@@ -0,0 +1,76 @@
+from typing import List, Protocol
+
+from autogen.agentchat import ConversableAgent
+
+from .document import Document
+from .graph_store import GraphStore
+
+
+class GraphRagAgent(ConversableAgent, Protocol):
+    """
+    A graph rag agent is a conversable agent which could query graph database for answers.
+
+    An implementing agent class would
+    1. create a graph in the underlying database with input documents
+    2. use the retrieve() method to retrieve information.
+    3. use the retrieved information to generate and send back messages.
+
+    For example,
+    graph_rag_agent = GraphRagAgent(
+        name="movie knowledge graph agent",
+        human_input_mode="NEVER",
+        max_consecutive_auto_reply=3,
+        retrieve_config={
+            "docs_path": [
+                "./data/movies.txt",
+            ],
+            "llm_config" = autogen.config_list_from_json("OAI_CONFIG_LIST")
+            "database_config" = {
+                "host": "127.0.0.1",
+                "port": 6379,
+                "table_name": "movies"
+            }
+        },
+    )
+
+    # initialize database (internally)
+    # self._init_db(input_doc=[Document(doc) for doc in retrieve_config["docs_path"]])
```

Comment:
Does this comment mean that the initialization happens when the GraphRagAgent is constructed?

Reply:
Not necessary. 
This step is to build graph in db with input documents.
We don't need this step if a graph is already built. We would only need to connect to the db.

Reply:
I understand that it doesn't always happen. My point is that this is not user's code and probably shouldn't appear here.

Reply:
IMAO, this is user code.
Graph DB is user specific choice and also the way to create the graph.
This is part of the initialization which implement the logic of creating the graph in the specific db.